### DISCUSSION TITLE
How do I add a custom config dir flag to my Angular Storybook project?

### DISCUSSION THREAD

**Discussion by stephh939** [storybookjs/storybook#27080] [Help] [answered]
### Summary

I need to tell my project where my main.js file is located at but I can't see to figure out how to add a custom config dir flag to Storybook. My project structure is in the additional information section along with where the `main.ts` file is. 

I saw in this discussion #24652 that "The -c flag is for a custom config directory where it can find your `main.js`". But again where/how do I use this -c flag?? 

There are multiple Nx console projects within my Angular "dbaasui" project and the specific project I am trying to add stories to is called shared-ui-components-v3.

If there is any other needed information please let me know and I'll provide it.

### Additional information

Project structure:
```
dbaasui
|_ _ apps
|     |_ _ unified-management-console
|          |_ _src
|              |_ _ main.ts                      <-- this is my "main" file I need to reference
|_ _libs
|    |_ _shared
|         |_ _ui-components-v3
|              |_ _.storybook
|                   |_ _main.ts
|                   |_ _preview.ts
|                   |_ _tsconfig.json
|              |_ _src
|              |_ _project.json
|              |_ _package.json
|              |_ _tsconfig.json
|
```

**Comment by Marklb** [comment, marked as answer]
For an Angular project, you can specify a path in the builder config's `configDir` property, in `angular.json`(*or Nx equivalent*).
```json
"storybook": {
  "builder": "@storybook/angular:start-storybook",
  "options": {
  	"configDir": ".storybook",
    "browserTarget": "shared-ui-components-v3:build",
    "port": 6006,
	"compodoc": true,
	"compodocArgs": [ "-e", "json", "-d", "." ]
  }
},
"build-storybook": {
  "builder": "@storybook/angular:build-storybook",
  "options": {
  	"configDir": ".storybook",
    "browserTarget": "shared-ui-components-v3:build",
    "outputDir": "storybook-static",
	"compodoc": true,
	"compodocArgs": [ "-e", "json", "-d", "." ]
  }
},
```
*I think the property is already in the config added by the Storybook init script.*

If you are trying to replace only the `main.ts` then I don't think that is supported. Typically each project will have it's own `.storybook` directory.
